File: DotNetCode/Controllers/ShiftController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using OfficeConnect_Web.Models;
using OfficeConnect_Web.ViewModel;

namespace OfficeConnect_Web.Controllers
{
    [AuthAttribute]
    public class ShiftController : Controller
    {
        ShiftModel SM = new ShiftModel();

        // POST: Shift/GetAllShift
        [Route("Shift/GetAllShift")]
        [HttpPost]
        public ActionResult GetAllShift(ShiftMasterViewModel model)
        {
            try
            {
                var emp = SM.GetAllShift(model);
                return Json(emp, JsonRequestBehavior.AllowGet);
            }
            catch (CustomApiException ex)
            {
                return Json(new { StatusCode = ex.StatusCode, Message = ex.Message });
            }
        }
        // POST: Shift/GetShift
        [Route("Shift/GetShift")]
        [HttpPost]
        public ActionResult GetShift(ShiftMasterViewModel model)
        {
            try
            {
                var emp = SM.GetShift(model);
                return Json(emp, JsonRequestBehavior.AllowGet);
            }
            catch (CustomApiException ex)
            {
                return Json(new { StatusCode = ex.StatusCode, Message = ex.Message });
            }
        }
        // POST: Shift/AddShift
        [Route("Shift/AddShift")]
        [HttpPost]
        public ActionResult AddShift(ShiftMasterViewModel model)
        {
            try
            {
                var emp = SM.AddShift(model);
                return Json(emp, JsonRequestBehavior.AllowGet);
            }
            catch (CustomApiException ex)
            {
                return Json(new { StatusCode = ex.StatusCode, Message = ex.Message });
            }
        }
        // POST: Shift/UpdateShift
        [Route("Shift/UpdateShift")]
        [HttpPost]
        public ActionResult UpdateShift(ShiftMasterViewModel model)
        {
            try
            {
                var emp = SM.UpdateShift(model);
                return Json(emp, JsonRequestBehavior.AllowGet);
            }
            catch (CustomApiException ex)
            {
                return Json(new { StatusCode = ex.StatusCode, Message = ex.Message });
            }
        }
        // POST: Shift/DeleteShift
        [Route("Shift/DeleteShift")]
        [HttpPost]
        public ActionResult DeleteShift(ShiftMasterViewModel model)
        {
            try
            {
                var emp = SM.DeleteShift(model);
                return Json(emp, JsonRequestBehavior.AllowGet);
            }
            catch (CustomApiException ex)
            {
                return Json(new { StatusCode = ex.StatusCode, Message = ex.Message });
            }
        }
        // POST: Shift/GetAllShiftGrouping
        [Route("Shift/GetAllShiftGrouping")]
        [HttpPost]
        public ActionResult GetAllShiftGrouping(ShiftGroupingViewModel model)
        {
            try
            {
                var emp = SM.GetAllShiftGrouping(model);
                return Json(emp, JsonRequestBehavior.AllowGet);
            }
            catch (CustomApiException ex)
            {
                return Json(new { StatusCode = ex.StatusCode, Message = ex.Message });
            }
        }
        ////// POST: Shift/LocationShiftGrouping
        ////[Route("Shift/LocationShiftGrouping")]
        ////[HttpPost]
        ////public ActionResult LocationShiftGrouping(ShiftGroupingViewModel model)
        ////{
        ////    try
        ////    {
        ////        var emp = SM.LocationShiftGrouping(model);
        ////        return Json(emp, JsonRequestBehavior.AllowGet);
        ////    }
        ////    catch (CustomApiException ex)
        ////    {
        ////        return Json(new { StatusCode = ex.StatusCode, Message = ex.Message });
        ////    }
        ////}
        
        // POST: Shift/AddShiftGrouping
        [Route("Shift/AddShiftGrouping")]
        [HttpPost]
        public ActionResult AddShiftGrouping(ShiftGroupingViewModel model)
        {
            try
            {
                var emp = SM.AddShiftGrouping(model);
                return Json(emp, JsonRequestBehavior.AllowGet);
            }
            catch (CustomApiException ex)
            {
                return Json(new { StatusCode = ex.StatusCode, Message = ex.Message });
            }
        }
        // POST: Shift/DDShift
        [Route("Shift/DDShift")]
        [HttpPost]
        public ActionResult DDShift(ShiftGroupingViewModel Empdd)
        {
            try
            {
                var Emp = SM.DDShift(Empdd);
                return Json(Emp, JsonRequestBehavior.AllowGet);
            }
            catch (CustomApiException ex)
            {
                return Json(new { StatusCode = ex.StatusCode, Message = ex.Message });
            }
        }
        // POST: Shift/GetAllShiftEmployee
        [Route("Shift/GetAllShiftEmployee")]
        [HttpPost]
        public ActionResult GetAllShiftEmployee(ShiftEmployeeMasterViewModel model)
        {
            try
            {
                var emp = SM.GetAllShiftEmployee(model);
                return Json(emp, JsonRequestBehavior.AllowGet);
            }
            catch (CustomApiException ex)
            {
                return Json(new { StatusCode = ex.StatusCode, Message = ex.Message });
            }
        }
        // POST: Shift/AddShiftEmployee
        [Route("Shift/AddShiftEmployee")]
        [HttpPost]
        public ActionResult AddShiftEmployee(ShiftEmployeeMappingMasterViewModel model)
        {
            try
            {
                var emp = SM.AddShiftEmployee(model);
                return Json(emp, JsonRequestBehavior.AllowGet);
            }
            catch (CustomApiException ex)
            {
                return Json(new { StatusCode = ex.StatusCode, Message = ex.Message });
            }
        }
        // POST: Shift/RemoveShiftEmployee
        [Route("Shift/RemoveShiftEmployee")]
        [HttpPost]
        public ActionResult RemoveShiftEmployee(ShiftEmployeeMappingMasterViewModel model)
        {
            try
            {
                var emp = SM.RemoveShiftEmployee(model);
                return Json(emp, JsonRequestBehavior.AllowGet);
            }
            catch (CustomApiException ex)
            {
                return Json(new { StatusCode = ex.StatusCode, Message = ex.Message });
            }
        }
    }
}

File: java/src/main/java/com/officeconnect/controller/ShiftController.java
package com.officeconnect.controller;

import com.officeconnect.dto.*;
import com.officeconnect.service.ShiftService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/Shift")
public class ShiftController {

    @Autowired
    private ShiftService shiftService;

    @PostMapping("/GetAllShift")
    public ResponseEntity<?> getAllShift(@RequestBody ShiftMasterViewModel model) {
        try {
            List<ShiftMasterViewModel> result = shiftService.getAllShift(model);
            return ResponseEntity.ok(result);
        } catch (RuntimeException ex) {
            return ResponseEntity.status(HttpStatus.NOT_FOUND)
                .body(Map.of("StatusCode", 404, "Message", ex.getMessage()));
        }
    }

    @PostMapping("/AddShift")
    public ResponseEntity<?> addShift(@RequestBody ShiftMasterViewModel model) {
        try {
            ShiftMasterViewModel result = shiftService.addShift(model);
            return ResponseEntity.ok(result);
        } catch (RuntimeException ex) {
            return ResponseEntity.status(HttpStatus.NOT_FOUND)
                .body(Map.of("StatusCode", 404, "Message", ex.getMessage()));
        }
    }

    @PostMapping("/UpdateShift")
    public ResponseEntity<?> updateShift(@RequestBody ShiftMasterViewModel model) {
        try {
            ShiftMasterViewModel result = shiftService.updateShift(model);
            return ResponseEntity.ok(result);
        } catch (RuntimeException ex) {
            return ResponseEntity.status(HttpStatus.NOT_FOUND)
                .body(Map.of("StatusCode", 404, "Message", ex.getMessage()));
        }
    }

    @PostMapping("/DeleteShift")
    public ResponseEntity<?> deleteShift(@RequestBody ShiftMasterViewModel model) {
        try {
            ShiftMasterViewModel result = shiftService.deleteShift(model);
            return ResponseEntity.ok(result);
        } catch (RuntimeException ex) {
            return ResponseEntity.status(HttpStatus.NOT_FOUND)
                .body(Map.of("StatusCode", 404, "Message", ex.getMessage()));
        }
    }

    @PostMapping("/GetShift")
    public ResponseEntity<?> getShift(@RequestBody ShiftMasterViewModel model) {
        try {
            List<ShiftMasterViewModel> result = shiftService.getShift(model);
            return ResponseEntity.ok(result);
        } catch (RuntimeException ex) {
            return ResponseEntity.status(HttpStatus.NOT_FOUND)
                .body(Map.of("StatusCode", 404, "Message", ex.getMessage()));
        }
    }

    @PostMapping("/GetAllShiftGrouping")
    public ResponseEntity<?> getAllShiftGrouping(@RequestBody ShiftGroupingMasterViewModel model) {
        try {
            List<ShiftGroupingMasterViewModel> result = shiftService.getAllShiftGrouping(model);
            return ResponseEntity.ok(result);
        } catch (RuntimeException ex) {
            return ResponseEntity.status(HttpStatus.NOT_FOUND)
                .body(Map.of("StatusCode", 404, "Message", ex.getMessage()));
        }
    }

    @PostMapping("/AddShiftGrouping")
    public ResponseEntity<?> addShiftGrouping(@RequestBody ShiftGroupingMasterViewModel model) {
        try {
            ShiftGroupingMasterViewModel result = shiftService.addShiftGrouping(model);
            return ResponseEntity.ok(result);
        } catch (RuntimeException ex) {
            return ResponseEntity.status(HttpStatus.NOT_FOUND)
                .body(Map.of("StatusCode", 404, "Message", ex.getMessage()));
        }
    }

    @PostMapping("/DDShift")
    public ResponseEntity<?> ddShift(@RequestBody ShiftGroupingMasterViewModel model) {
        try {
            List<ShiftMasterViewModel> result = shiftService.ddShift(model);
            return ResponseEntity.ok(result);
        } catch (RuntimeException ex) {
            return ResponseEntity.status(HttpStatus.NOT_FOUND)
                .body(Map.of("StatusCode", 404, "Message", ex.getMessage()));
        }
    }

    @PostMapping("/CreateShift")
    public ResponseEntity<?> createShift(@RequestBody ShiftMasterViewModel model) {
        try {
            List<ShiftMasterViewModel> result = shiftService.createShift(model);
            return ResponseEntity.ok(result);
        } catch (RuntimeException ex) {
            return ResponseEntity.status(HttpStatus.NOT_FOUND)
                .body(Map.of("StatusCode", 404, "Message", ex.getMessage()));
        }
    }

    @PostMapping("/GetShiftByEmployee")
    public ResponseEntity<?> getShiftByEmployee(@RequestBody ShiftMasterViewModel model) {
        try {
            List<ShiftMasterViewModel> result = shiftService.getShiftByEmployee(model);
            return ResponseEntity.ok(result);
        } catch (RuntimeException ex) {
            return ResponseEntity.status(HttpStatus.NOT_FOUND)
                .body(Map.of("StatusCode", 404, "Message", ex.getMessage()));
        }
    }

    @PostMapping("/AssignShift")
    public ResponseEntity<?> assignShift(@RequestBody ShiftMasterViewModel model) {
        try {
            ShiftMasterViewModel result = shiftService.assignShift(model);
            return ResponseEntity.ok(result);
        } catch (RuntimeException ex) {
            return ResponseEntity.status(HttpStatus.NOT_FOUND)
                .body(Map.of("StatusCode", 404, "Message", ex.getMessage()));
        }
    }

    @PostMapping("/AddShiftEmployee")
    public ResponseEntity<?> addShiftEmployee(@RequestBody Map<String, Object> model) {
        try {
            Map<String, Object> result = shiftService.addShiftEmployee(model);
            return ResponseEntity.ok(result);
        } catch (RuntimeException ex) {
            return ResponseEntity.status(HttpStatus.NOT_FOUND)
                .body(Map.of("StatusCode", 404, "Message", ex.getMessage()));
        }
    }

    @PostMapping("/GetAllShiftEmployee")
    public ResponseEntity<?> getAllShiftEmployee(@RequestBody Map<String, Object> model) {
        try {
            ShiftEmployeeListViewModel result = shiftService.getAllShiftEmployee(model);
            return ResponseEntity.ok(result);
        } catch (RuntimeException ex) {
            return ResponseEntity.status(HttpStatus.NOT_FOUND)
                .body(Map.of("StatusCode", 404, "Message", ex.getMessage()));
        }
    }

    @PostMapping("/RemoveShiftEmployee")
    public ResponseEntity<?> removeShiftEmployee(@RequestBody Map<String, Object> model) {
        try {
            Map<String, Object> result = shiftService.removeShiftEmployee(model);
            return ResponseEntity.ok(result);
        } catch (RuntimeException ex) {
            return ResponseEntity.status(HttpStatus.NOT_FOUND)
                .body(Map.of("StatusCode", 404, "Message", ex.getMessage()));
        }
    }

    @PostMapping("/LocationShiftGrouping")
    public ResponseEntity<?> locationShiftGrouping(@RequestBody Map<String, Object> model) {
        try {
            List<Map<String, Object>> result = shiftService.locationShiftGrouping(model);
            return ResponseEntity.ok(result);
        } catch (RuntimeException ex) {
            return ResponseEntity.status(HttpStatus.NOT_FOUND)
                .body(Map.of("StatusCode", 404, "Message", ex.getMessage()));
        }
    }
}